pandas/core/indexes/base.py

```diff
@@ -3683,7 +3683,16 @@ def join(self, other, how="left", level=None, return_indexers=False, sort=False)
                 return self._join_non_unique(
                     other, how=how, return_indexers=return_indexers
                 )
-        elif self.is_monotonic and other.is_monotonic:
+        elif (
+            self.is_monotonic
+            and other.is_monotonic
+            and (
+                not isinstance(self, ABCMultiIndex)
+                or not any(is_categorical_dtype(dtype) for dtype in self.dtypes)
+            )
+        ):
+            # Categorical is monotonic if data are ordered as categories, but join can
+            #  not handle this in case of not alphabetically monotonic GH#38502
```

Comment:
is alphabetical really the issue?  what about numeric or dt64 Categoricals?

Reply:
Hm bad wording probably. If the object is lexicographically ordered everything is fine. Monotonic in the sense of categorical monotonic is the problem

Reply:
I will add tests for different types of Categorical.s

Reply:
Added float, int and timestamps